libxrdp/xrdp_rdp.c

```diff
@@ -820,12 +823,30 @@ xrdp_rdp_incoming(struct xrdp_rdp *self)
     self->mcs_channel = self->sec_layer->mcs_layer->userid +
                         MCS_USERCHANNEL_BASE;
     DEBUG(("out xrdp_rdp_incoming mcs channel %d", self->mcs_channel));
-    g_strncpy(self->client_info.client_addr,
-              self->sec_layer->mcs_layer->iso_layer->trans->addr,
+    g_strncpy(self->client_info.client_addr, iso->trans->addr,
               sizeof(self->client_info.client_addr) - 1);
-    g_strncpy(self->client_info.client_port,
-              self->sec_layer->mcs_layer->iso_layer->trans->port,
+    g_strncpy(self->client_info.client_port, iso->trans->port,
               sizeof(self->client_info.client_port) - 1);
+
+    /* log TLS version and cipher when TLS is used */
+    /* TODO: client_addr, client_port is empty when IPv6 enabled */
+
+    if (iso->selectedProtocol > PROTOCOL_RDP)
+    {
+        log_message(LOG_LEVEL_INFO,
+                    "TLS connection established from %s: %s with cipher %s",
```

Comment:
As described in #422, I was inspired by postfix. The log format is brought from postfix.  I like it because protocol name is put beside cipher. Yours is not bad but protocol name and cipher are a little bit distant for me.

I agree with logging the port.  The IP address part could be IPv6.  `127.0.0.1 port 5678` `::1 port 5678` like other logs would be fine.